media/live-seekable-range.js

```diff
@@ -1,68 +1,88 @@
-const video = document.createElement('video');
+const video = document.querySelector('video');
 
 const mediaSource = new MediaSource();
 video.src = URL.createObjectURL(mediaSource);
   
 mediaSource.addEventListener('sourceopen', function() {
   URL.revokeObjectURL(video.src);
 
-  const videoUrl = 'https://storage.googleapis.com/media-session/sample.webm';
-  const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
-
-  // Makes video behave like a live stream.
-  mediaSource.duration = +Infinity;
+  mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
 
   log('Fetching video init segment...');
-  fetch(videoUrl, { headers: { range: 'bytes=0-299' } })
-  .then(response => response.arrayBuffer())
-  .then(data => {
-
-    // Append video init segment.
-    sourceBuffer.appendBuffer(data);
-    sourceBuffer.addEventListener('updateend', function() {
-      logTimeRanges();
-
-      log('Fetching video segment that starts at 3 seconds...');
-      fetch(videoUrl, { headers: { range: 'bytes=567140-1196488' } })
-      .then(response => response.arrayBuffer())
-      .then(data => {
+  fetchAndAppendMediaSegment('bytes=0-299');
+});
 
-        // Append video segment that starts at 3 seconds.
-        sourceBuffer.appendBuffer(data);
-        sourceBuffer.addEventListener('updateend', function() {
-          logTimeRanges();
-        }, { once: true });
+function onSetInfiniteDurationButtonClick() {
+  log('User clicked "mediaSource.duration = +Infinity" button');
+  // Makes video behave like a live stream.
+  mediaSource.duration = +Infinity;
+  logMediaInfo();
+}
 
-      });
-    }, { once: true });
-  });
-});
+function onFetchAndAppend3To6MediaButtonClick() {
+  log('Fetching video segment that starts at 3 seconds...');
+  fetchAndAppendMediaSegment('bytes=567140-1196488');
+}
 
-function onSetLiveSeekableRangeButtonClick() {
-  log('User clicked "mediaSource.setLiveSeekableRange(40, 42)" button');
-  mediaSource.setLiveSeekableRange(40 /* start */, 42 /* end */);
-  logTimeRanges();
+function onSetLiveSeekableRangeButtonClick(event) {
+  const re = /mediaSource\.setLiveSeekableRange\((\d+), (\d+)\)/;
+  const start = Number(re.exec(event.target.textContent)[1]);
+  const end = Number(re.exec(event.target.textContent)[2]);
+  log(`User clicked "mediaSource.setLiveSeekableRange(${start}, ${end})" button`);
+  mediaSource.setLiveSeekableRange(start, end);
```

Comment:
Many of these operations depend on the mediaSource being in the 'open' state.
So, if a sequence contains endOfStream() without a subsequent fetchAndAppend, then exception will occur, like "Uncaught InvalidStateError: Failed to set the 'duration' property on 'MediaSource': The MediaSource's readyState is not 'open'. (Your browser may not support this feature.)" for many of the other buttons.

Suggestion is to at least move that exception message to be in-line in the log with the sequence of operations.

Second suggestion is to catch the exception in the button click handler functions, so that they can still logMediaInfo after (and so they perhaps can give a better understanding of what's going on to the user).